internal/cli/plugin/first_class.go

```diff
@@ -0,0 +1,143 @@
+// Copyright 2024 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package plugin
+
+import (
+	"fmt"
+
+	"github.com/google/go-github/v61/github"
+	"github.com/mongodb/mongodb-atlas-cli/atlascli/internal/plugin"
+	"github.com/spf13/cobra"
+)
+
+const (
+	SourceType = "first-class"
+)
+
+// uncomment example plugin to test first class plugin feature.
+var firstClassPlugins = []*FirstClassPlugin{
+	// {
+	// 	Name: "atlas-cli-first-class-plugin-example",
+	// 	Github: &Github{
+	// 		Owner: "stefan4h",
+	// 		Name: "atlas-cli-first-class-plugin-example",
+	// 	},
+	// 	Commands: []*Command{
+	// 		{
+	// 			Name: "first-class",
+	// 			Description: "Root command of the Atlas CLI first class plugin example",
+	// 		},
+	// 	},
+	// },
+}
+
+type Command struct {
+	Name        string
+	Description string
+}
+
+type Github struct {
+	Owner string
+	Name  string
+}
+
+type FirstClassPlugin struct {
+	Name     string
+	Github   *Github
+	Commands []*Command
+}
+
+func IsFirstClassPluginCmd(cmd *cobra.Command) bool {
+	if cmdSourceType, ok := cmd.Annotations["sourceType"]; ok && cmdSourceType == SourceType {
```

Comment:
can we make these annotations consts? `sourceType`, `sourcePluginName`, etc

Reply:
done